### DISCUSSION TITLE
pnpm not recognized after I installed

### DISCUSSION THREAD

**Discussion by yachji** [coreybutler/nvm-windows#1015] [Support]
I downloaded [nvm-noinstall.zip](https://github.com/coreybutler/nvm/releases) and installed nvm manually.then I installed pnpm by `npm install pnpm -g` and it successed. But when I type `pnpm` command ,i got  `pnpm is not an internal or external command`.
How can I solve the problem?

**Comment by coreybutler** [comment]
I attempted to install pnpm on Node 18.16.1 with no trouble:

<img width="854" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/d43c4e57-8adb-46d6-81f1-beac3c464fcb">

So, I suspect there may be a missing detail. Did you restart your terminal? Did you run `nvm debug` to see if there are any detected issues? I assume you ran `nvm use x.x.x` where `x.x.x` is the version you want, otherwise the npm installation would have failed.

The bottom line: there are too many unknown details to be able to diagnose the problem. If you can be more specific, I might be able to offer better help. You could start by posting the results of `nvm debug`.

**Comment by yachji** [comment]
I install nvm after https://github.com/coreybutler/nvm-windows/wiki#manual-installation
then i execute `nvm install lts`
`nvm debug` 
![image](https://github.com/coreybutler/nvm-windows/assets/25786615/80c7c152-1245-4f34-aad3-e62d0558f14c)

![image](https://github.com/coreybutler/nvm-windows/assets/25786615/9be90c80-8106-49f1-87fc-e3e72e100752)

`pnpm -v`
it still cann't find pnpm  
@coreybutler

**Reply by coreybutler** [reply]
What else was in the nvm debug output? What were the results of the permissions check, i.e. 
<img width="428" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/0b53513b-1bec-4032-a7b8-9f69a4f640f3">

Were there any issues with the terminal or other problems listed, such as
<img width="851" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/1ea23dc8-16b4-4861-8d78-c0c73f11a58e">

To see if pnpm is properly installed, go to `D:\MyComponents\nvm-noinstall-v1.1.11\v18.17.1`. It should look similar to this (specifically the highlighted files):

<img width="618" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/5bc2ba79-6785-4c2e-8817-f75edc8e7ab7">

Then click on `node_modules`. This directory should have `pnpm` in it.

<img width="520" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/55714dcf-1109-418d-8300-3a6290d22d04">

That folder should contain additional files like this:

<img width="518" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/00630c3d-a76b-4aa8-a091-5a8d57d02121">

Assuming the above is correct on your computer, you should be able to run `D:\MyComponents\nvm-noinstall-v1.1.11\pnpm.cmd -v` to see the version:

<img width="500" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/8731dade-f5f8-41ac-9cc8-5096814603be">

If this works, then pnpm is installed correctly and something is blocking it. It's not uncommon for Windows to silently block scripts it thinks are insecure, so be sure to check your security settings to assure it's not being blocked.
